chore(ci): use github actions

RobinKnipe opened the pull request:
Handle basic code quality assurance checking (CI) with GitHub Actions.

RobinKnipe committed:
chore(ci): use github actions

Handle basic code quality assurance checking (CI) with GitHub Actions.

daniel-ac-martin commented on .github/workflows/node.js.yml:
```diff
@@ -0,0 +1,31 @@
+
+on:
+  push:
+    branches: [ master ]
```
Are you sure you want this clause?

daniel-ac-martin replied:
i.e. Why not pushes to all branches?

RobinKnipe replied:
Agreed, fixed

daniel-ac-martin commented on .github/workflows/node.js.yml:
```diff
@@ -0,0 +1,31 @@
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x]
```
We should probably move to v12 and forget about v10.

However we would need to migrate to a v12 base-image for that.

RobinKnipe replied:
Switched to use `node12` only. The recent changes (rebased) introduce a node 12 container, so we should be good...

daniel-ac-martin replied:
It's nice to at least unit test on 14 as well. They are quick to run and they might flag up future problems.

daniel-ac-martin replied:
But that can be sorted in another PR.

RobinKnipe replied:
Fixed

daniel-ac-martin commented on .github/workflows/node.js.yml:
```diff
@@ -0,0 +1,31 @@
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm ci
+    - run: npm run build --if-present
```
I'd remove this. - We don't have a `build` script and if we did it wouldn't necessarily be a prerequisite of `npm test`.

RobinKnipe replied:
Agreed, removed

RobinKnipe committed:
chore(ci): recommendations from peer review

Implements PR suggestions:
 - drop the node 10 matrix build
 - run the CI actions for all pushes and PRs
 - remove the unnecessary `npm build` job

RobinKnipe committed:
fix(tests): separate mocks and tests

When Mocha runs on GitHub Actions, it tries to run the mock files to
look for tests... this triggers a syntax error. Instead, separate the
tests into their own directory, so that they can be treated in isolation
by Mocha.

RobinKnipe committed:
fix(testing): add missing config resources

Ensure all the necessary configuration files are copied into the docker
test image, so that testing runs correctly.

daniel-ac-martin commented on .github/workflows/node.js.yml:
```diff
@@ -0,0 +1,21 @@
+
+name: Node.js CI
+
+on: [push, pull_request  ]
```
Whitespace is a bit funky here.

RobinKnipe replied:
Fixed

daniel-ac-martin reviewed (dismissed):
Good enough to merge but check my comments and, if you agree, either fix or add them as issues to be fixed later.
